src/parse_instruction.py

```diff
@@ -278,15 +296,13 @@ def constant_fold(arg: Argument) -> Argument:
     return arg
 
 
-def replace_bare_reg(arg: Argument, arch: ArchAsmParsing) -> Argument:
+def replace_bare_reg(arg: Argument, arch: ArchAsmParsing, used_reg_names: UsedRegNames) -> Argument:
     """If `arg` is an AsmGlobalSymbol whose name matches a known or aliased register,
     convert it into a Register and return it. Otherwise, return the original `arg`."""
     if isinstance(arg, AsmGlobalSymbol):
         maybe_reg = Register(arg.symbol_name)
-        if maybe_reg in arch.aliased_regs:
-            return arch.aliased_regs[maybe_reg]
-        if maybe_reg in arch.all_regs:
-            return maybe_reg
+        if maybe_reg in arch.all_regs or maybe_reg.register_name in arch.aliased_regs:
+            return used_reg_names.input_to_internal(maybe_reg.register_name, arch)
```

Comment:
```suggestion
        reg_name = arg.symbol_name
        if Register(reg_name) in arch.all_regs or reg_name in arch.aliased_regs:
            return used_reg_names.input_to_internal(reg_name, arch)
```
to get rid of maybe_reg